### PR TITLE
Geo map fixes

### LINKED ISSUE #55: Geo Map: hide Jittering when Cluster points is on
Jittering is useless if Cluster points is on (because clustering sums points into clusters). Turn it off then Cluster points is selected.

### PR DESCRIPTION
**Pull request by matejklemen**
##### Issue
Fixes #55 and (partially?) #69.

##### Description of changes
Regarding #55: prevents change of jittering and shadows its value when 'Cluster points' is checked.  

Regarding #69: plots missing values for **discrete** variables with a gray color (somewhat arbitrarily chosen, please feel free to suggest an alternative, if it does not fit the color scheme). The handling of missing values is local to this widget and does not change any of the attribute's properties.  
In a nutshell, nans get assigned a new (previously unassigned) integer, which maps to gray color.  

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ajdapretnar** [comment]
I like this. The only comment I would have is that NA in the legend seems slightly redundant to me. Elsewhere in Orange, gray is simply missing and it does not appear in the legend. I think it makes sense to have the same principle here.

Also, @robertcv could you please check how this is compatible with the changes you plan to introduce?

File: orangecontrib/geo/widgets/owmap.py [hunk 1]
```diff
@@ -263,12 +263,28 @@ def set_marker_color(self, attr, update=True):
...
                 _values = np.asarray(self.data.domain[attr].values)
-                __values = self.data.get_column_view(variable)[0].astype(np.uint16)
+
+                encoded_values = self.data.get_column_view(variable)[0]
+                is_na = np.isnan(encoded_values)
+                contains_na = np.any(is_na)
+
+                # Add a NA value + gray color locally (does not change attribute properties)
+                if contains_na:
+                    _values = np.append(_values, "NA")
```
Comment (robertcv) [inline orangecontrib/geo/widgets/owmap.py]: As @ajdapretnar said, no need for "NA"

File: orangecontrib/geo/widgets/owmap.py [hunk 2]
```diff
@@ -263,12 +263,28 @@ def set_marker_color(self, attr, update=True):
...
+                    used_colors = np.vstack((variable.colors, [186, 189, 182]))
```
Comment (robertcv) [inline orangecontrib/geo/widgets/owmap.py]: In Scatter Plot we use [128, 128, 128] for gray and it would be nice to be consistent with it.

**Comment by matejklemen** [comment]
Both things fixed.  
The remaining "NA" in code is the one that appears in tooltip when a marker is hovered-over.

**Comment by robertcv** [comment]
Please also replace "NA" in the tooltip with "?".

**Comment by matejklemen** [comment]
Fixed as well
